airbyte-server/src/test/java/io/airbyte/server/handlers/ConnectionsHandlerTest.java

```diff
@@ -612,6 +619,321 @@ void testEnumConversion() {
       assertTrue(Enums.isCompatible(NamespaceDefinitionType.class, io.airbyte.config.JobSyncConfig.NamespaceDefinitionType.class));
     }
 
+    @Test
+    void testNoDiff() {
+      final AirbyteStreamConfiguration streamConfiguration1 = getStreamConfiguration(List.of("cursor1"), List.of(List.of("pk1")), SyncMode.INCREMENTAL,
+          DestinationSyncMode.APPEND_DEDUP);
+
+
+      final AirbyteStreamConfiguration streamConfiguration2 = getStreamConfiguration(List.of("cursor2"), List.of(List.of("pk2")),
+          SyncMode.FULL_REFRESH,
+          DestinationSyncMode.OVERWRITE);
+
+      final AirbyteCatalog catalog1 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+      final AirbyteCatalog catalog2 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+
+      assertTrue(connectionsHandler.getConfigurationDiff(catalog1, catalog2).isEmpty());
+    }
+
+    @Test
+    void testNoDiffIfStreamAdded() {
+      final AirbyteStreamConfiguration streamConfiguration1 = getStreamConfiguration(List.of("cursor1"), List.of(List.of("pk1")),
+          SyncMode.INCREMENTAL,
+          DestinationSyncMode.APPEND_DEDUP);
+
+      final AirbyteStreamConfiguration streamConfiguration2 = getStreamConfiguration(List.of("cursor2"), List.of(List.of("pk2")),
+          SyncMode.FULL_REFRESH,
+          DestinationSyncMode.OVERWRITE);
+
+      final AirbyteCatalog catalog1 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1)
+              )
+          );
+      final AirbyteCatalog catalog2 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+
+      assertTrue(connectionsHandler.getConfigurationDiff(catalog1, catalog2).isEmpty());
+    }
+
+    @Test
+    void testCursorOrderDoesntMatter() {
```

Comment:
Are you sure that cursor order doesn't matter? I'm not totally sure what it means to have a list of cursor fields, but I had assumed that the order _did_ matter, e.g. it would first use the first cursor field to decide the order of records to read, and if multiple records had the same value for the cursor then it would use the next one in the list to tie-break, and so on

Reply:
I am not sure, I didn't find a way to select multiple cursor. It is clearer for the primary key (details in the comment bellow).